compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -59,30 +60,113 @@ fn eat_operand_keyword<'a>(
     }
 }
 
+/// A parsed list of attributes that is not attached to any item.
+/// Used to check whether `asm!` arguments are configured out.
+struct AsmAttrVec(ast::AttrVec);
+
+impl AsmAttrVec {
+    fn parse<'a>(ecx: &ExtCtxt<'a>, p: &mut Parser<'a>) -> PResult<'a, Self> {
+        let span_start = p.token.span;
+
+        let mut attributes = ast::AttrVec::new();
+        loop {
+            if p.token != token::Pound {
+                break;
+            }
+
+            let attr = p.parse_attribute(rustc_parse::parser::attr::InnerAttrPolicy::Permitted)?;
+            attributes.push(attr);
+        }
+
+        for attr in attributes.iter() {
+            match attr.name() {
+                Some(sym::cfg | sym::cfg_attr) => {
+                    if !ecx.ecfg.features.asm_cfg() {
+                        let span = span_start.to(p.prev_token.span);
+                        feature_err(ecx.sess, sym::asm_cfg, span, fluent::builtin_macros_asm_cfg)
+                            .emit();
+                    }
+                }
+                _ => {
+                    ecx.dcx().emit_err(errors::AsmAttributeNotSupported { span: attr.span() });
+                }
```

Comment:
non-`cfg` attributes are now accepted by the parser, but emit this error. Previously attributes were only parsed (and later rejected, unless `#[feature(stmt_expr_attributes)]` was enabled) on expressions. We now also always parse them on operands.